tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -52,28 +52,60 @@ def update_resource_tags(tag_action, resource, tags):
             if tag_action.manager.type == 'armresource':
                 raise NotImplementedError('Cannot tag generic ARM resources.')
```

Comment:
Ideally with this change you'll now be able to tag generic arm resources.  I'd remove this safety check and try running a policy on all resources using the `azure.armresource` resource type and see if it breaks them all or not :)  Especially Network Security Groups were breaking previously (Was deleting all rules).

Reply:
We are going to leave this if check in there for armresources in this PR. We decided to break up the work as this PR fixes issues with specific resource tagging. https://github.com/capitalone/cloud-custodian/issues/2637 will be addressed in a separate PR with a robust solution that will ensure tagging generic arm resources works across all current and future Azure services.